LFS-622: Create a "demo" runmode

IntegralProgrammer opened the pull request:
**This PR is dependent upon `LFS-630` and will therefore remain as a _draft_ until `LFS-630` is merged into `dev`**

This PR implements the LFS-622 feature described on https://phenotips.atlassian.net/browse/LFS-622 by adding a warning banner if LFS is started with the `demo` runmode.

To test:

- Start LFS normally with `java -jar distribution/target/lfs-0.1-SNAPSHOT.jar`. Everything should work as normal.
- Stop LFS and restart with the `demo` runmode
  - `rm -rf sling/`
  - `java -jar distribution/target/lfs-0.1-SNAPSHOT.jar -Dsling.run.modes=demo`
- Everything should also still work, but a orange warning banner should be _always_ present, except for the login screen

IntegralProgrammer committed:
LFS-622: Create a "demo" runmode
Add warning banner module

IntegralProgrammer committed:
LFS-622: Create a "demo" runmode
Make the warning banner orange

IntegralProgrammer committed:
LFS-622: Create a "demo" runmode
Package lfs-banner as an OSGi module so that it can be conditionally loaded in the 'demo' runMode

IntegralProgrammer committed:
LFS-622: Create a "demo" runmode
Fix naming convention for LFS Warning Banner

marta- commented on modules/warning-banner/src/main/frontend/package.json:
```diff
@@ -0,0 +1,43 @@
+    "name": "lfs-warning-banner",
+    "version": "0.1.0",
+    "description": "Warning banner resources for the LFS Project",
+    "author": "SickKids",
```
Almost out of scope for this PR, but we should update this everywhere.

marta- commented on modules/warning-banner/src/main/frontend/src/warningBanner.jsx:
```diff
@@ -0,0 +1,54 @@
+    <StyledAppBar position="sticky">
+      <Toolbar>
+        <IconButton edge="start" color="inherit">
+          <WarningIcon fontsize="large"/>
```
This causes a warning in the console and doesn't make the icon larger.
```suggestion
          <WarningIcon fontSize="large"/>
```

marta- commented on modules/warning-banner/src/main/frontend/src/warningBanner.jsx:
```diff
@@ -0,0 +1,54 @@
+export default function WarningBanner() {
+  const StyledAppBar = withStyles({
+    root: {
+      background: "#FFA500"
```
Very minor, but can we use 
```suggestion
      backgroundColor: theme.pallette.warning.main
```
instead?

marta- reviewed (changes requested):
Looks good, minor comments.

sashaandjic commented:
tested

sashaandjic commented:
![image](https://user-images.githubusercontent.com/23243244/96932280-a0c25780-148c-11eb-865c-cbde5806f820.png)

IntegralProgrammer committed:
LFS-622: Create a "demo" runmode
Misc updates

IntegralProgrammer committed:
LFS-622: Create a "demo" runmode
Minor improvements to warningBanner.jsx

sashaandjic reviewed (approved):
re-tested

IntegralProgrammer committed:
LFS-622: Create a "demo" runmode
Pass style prop to Warning Banner

IntegralProgrammer committed:
LFS-622: Create a "demo" runmode
Refactor by renaming TopVisualElement --> PageStart

IntegralProgrammer committed:
LFS-622: Create a "demo" runmode
Rename Content node to WarningBanner node and change description

sashaandjic commented:
retested

sashaandjic commented:
retested

marta- commented on modules/warning-banner/src/main/provisioning/model.txt:
```diff
@@ -0,0 +1,21 @@
+#  under the License.
+#
+
+# The LFS Variants management
```
Misplaced comment?

IntegralProgrammer replied:
Fixed :heavy_check_mark:

marta- reviewed (approved):
All good after the comment in modules/warning-banner/src/main/provisioning/model.txt is fixed

IntegralProgrammer committed:
LFS-622: Create a "demo" runmode
Fix misplaced comment

sdumitriu committed:
[misc] Better feature description

sdumitriu committed:
LFS-622: Create a "demo" runmode
Renamed the module from "warning-banner" to "demo-banner", since it only warns about one thing

sdumitriu reviewed (approved):
I renamed everything from "warning-banner" to "demo-banner".
